plugins/fcp_can_c/fcp_can_c/generator.py

```diff
@@ -123,3 +124,59 @@ def check_impl_size(
                 )
             else:
                 return Ok(())
+
+        @register(verifier, "struct")  # type: ignore
+        def check_field_id_clash(
+            self: Any, fcp: FcpV2, extension: Any
+        ) -> Result[Nil, FcpError]:
+            """Check for duplicate field IDs inside structs."""
+            for struct in fcp.structs:
+                seen_ids = set()
+                for field in struct.fields:
+                    if field.field_id in seen_ids:
+                        return error(
+                            f"Duplicate field ID @{field.field_id} in struct '{struct.name}'",
+                            node=field,
+                        )
+                    seen_ids.add(field.field_id)
+            return Ok(())
+
+        @register(verifier, "impl")  # type: ignore
+        def check_service_method_id_clash(
+            self: Any, fcp: FcpV2, extension: Any
+        ) -> Result[Nil, FcpError]:
+            """Check for duplicate method IDs inside services."""
+            for service in fcp.services:  # iterate all services
+                seen_ids = set()
+                for method in service.methods:
+                    if method.id in seen_ids:
+                        return error(
+                            f"Duplicate method ID @{method.id} in service '{service.name}' "
+                            f"(method '{method.name}')",
+                            node=method,
+                        )
+                    seen_ids.add(method.id)
+            return Ok(())
+
+        @register(verifier, "impl")  # type: ignore
+        def check_rpc_method_id_clash(
```

Comment:
Remove, method ids can be repeated across different services. Identification of a method is done via the (service_id, method_id) tuple.